include/deal.II/lac/iterative_inverse.h

```diff
@@ -129,10 +129,9 @@ inline
 void
 IterativeInverse<VectorType>::initialize(const MatrixType &m, const PreconditionerType &p)
 {
-  // dummy variable
-  VectorType *v = 0;
-  matrix = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(m, *v));
-  preconditioner = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(p, *v));
+  VectorType v;
+  matrix = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(m, v));
+  preconditioner = std_cxx11::shared_ptr<PointerMatrixBase<VectorType> > (new_pointer_matrix_base(p, v));
```

Comment:
I don't think that this is a bug: all of the overloads of `new_pointer_matrix_base` in `pointer_matrix.h` do not name the vector argument. I believe the original author included that parameter just so that we could get nice function overloading based on the correct vector type.